pkg/sanity/volume.go

```diff
@@ -0,0 +1,194 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/client"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+
+	"github.com/libopenstorage/openstorage/volume"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Volume [Volume Tests]", func() {
+	var (
+		restClient   *client.Client
+		volumedriver volume.VolumeDriver
+	)
+
+	BeforeEach(func() {
+		var err error
+		restClient, err = volumeclient.NewDriverClient(osdAddress, "nfs", volume.APIVersion, "")
+
+		Expect(err).ToNot(HaveOccurred())
+		//volumedriver = clusterclient.ClusterManager(restClient)
+		volumedriver = volumeclient.VolumeDriver(restClient)
+	})
+
+	AfterEach(func() {
+
+	})
+
+	Describe("Volume Create", func() {
+
+		var (
+			volumerequest *api.VolumeCreateRequest
+			volumeID      string
+		)
+
+		BeforeEach(func() {
+
+			var err error
+
+			volumerequest = &api.VolumeCreateRequest{
+				Locator: &api.VolumeLocator{
+					Name: "vol-osd-sanity-cd",
+					VolumeLabels: map[string]string{
+						"class": "sanity-test-class",
+					},
+				},
+				Source: &api.Source{},
+				Spec: &api.VolumeSpec{
+					Size:   1,
+					Format: api.FSType_FS_TYPE_NFS,
+				},
+			}
+
+			volumeID, err = volumedriver.Create(volumerequest.GetLocator(), volumerequest.GetSource(), volumerequest.GetSpec())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(volumeID).ToNot(BeNil())
+		})
+
+		AfterEach(func() {
+			err := volumedriver.Delete(volumeID)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("Volume should be present in inspect", func() {
+
+			inspectVolumes := []string{volumeID}
+
+			By("only one volume to be returned and having the same name as the created volume in beforeeach.")
+			volumesList, err := volumedriver.Inspect(inspectVolumes)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(volumesList).NotTo(BeEmpty())
+			Expect(len(volumesList)).Should(BeEquivalentTo(1))
+			Expect(volumesList[0].GetId()).Should(BeEquivalentTo(volumeID))
+		})
+
+		It("Volume should be created with correct input data", func() {
+
+			inspectVolumes := []string{volumeID}
+
+			By("inspecting the details of the volume")
+			volumesList, err := volumedriver.Inspect(inspectVolumes)
+
+			volume := volumesList[0]
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(volume.GetSpec().GetSize()).To(BeEquivalentTo(1))
+			Expect(volume.GetSpec().GetFormat()).To(BeEquivalentTo(api.FSType_FS_TYPE_NFS))
+
+		})
+
+		It("should say volume already exists", func() {
+			By("Creating the volume with same name again")
+
+			_, err := volumedriver.Create(volumerequest.GetLocator(), volumerequest.GetSource(), volumerequest.GetSpec())
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("already exists"))
```

Comment:
I'm not sure what "Should" does or how it is different to `To`, but could you change it to `To`?